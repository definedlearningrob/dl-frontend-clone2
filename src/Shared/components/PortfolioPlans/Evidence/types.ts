import { EVIDENCE_CONTEXT_KIND, EVIDENCE_KIND } from '@shared/resources/enums';

export type ManualAddedEvidenceRecord = {
  itemId: string | null;
  itemType: Omit<EVIDENCE_CONTEXT_KIND, 'RUBRIC_GRADE'>;
};

export type EvidenceRecord = {
  itemId: string;
  itemType: EVIDENCE_KIND;
};
