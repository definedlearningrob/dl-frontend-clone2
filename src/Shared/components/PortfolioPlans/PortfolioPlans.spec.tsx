import { Route } from 'react-router-dom';
import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import { screen, within } from '@testing-library/dom';
import userEvent from '@testing-library/user-event';

import { studentInfoMock } from '@dc/tests/mocks/studentMocks';
import { userInfoMock } from '@dc/tests/mocks/userMocks';
import { renderWithRouterAndReduxProvider } from '@dc/utils/test';

import PORTFOLIO_PLANS_QUERY, {
  TPortfolioPlansData,
} from '@shared/graphql/student/query/portfolioPlans';
import PLAN_WITH_EVALUATION_QUERY from '@shared/graphql/student/query/planWithEvaluation';
import { EVALUATION_RESULTS_VALUES } from '@shared/resources/enums';
import { NavigationContextProvider } from '@shared/components/Sidebar/useNavigation';
import STUDENT_PORTFOLIO_PLANS from '@shared/graphql/user/query/studentPortfolioPlans';
import STUDENT_PORTFOLIO_PLAN_WITH_EVALUATION from '@shared/graphql/user/query/studentPortfolioPlanWithEvaluation';

import { StudentInfo, UserInfo } from './types';
import { PortfolioPlans } from './PortfolioPlans';

const studentPortfolioPlansWithEvaluationMock: MockedResponse = {
  request: {
    query: STUDENT_PORTFOLIO_PLAN_WITH_EVALUATION,
    variables: { uuid: '123', id: '48' },
  },
  result: {
    data: {
      student: {
        portfolio: {
          plan: {
            description:
              '<h5>Kentucky Grade 9 Individual Learning Plan</h5>\n<p>Congratulations! You are entering your first year of high school. This year should be an exciting time for you. Use this year to explore your likes and your dislikes and find a career that may be compatible with your interests and your strengths. Develop relationships with fellow classmates, teachers, school counselors and administrators.</p>\n<p>&nbsp;</p>\n<p>Try many clubs and organizations such as the speech club, soccer, a career and technical student organization related to your career pathway or any other extracurricular activity that matches your interests and strengths. Use this year as your launch pad to a successful high school career. Remember to have fun.</p>',
            evaluation: {
              id: '2570',
              __typename: 'Evaluation',
            },
            groups: [
              {
                description: '<p>KY ILP Grade 9 Indicators for Career Goals</p>',
                displayName: 'Career Goals',
                id: '119',
                name: 'Career Goals - KY Grade 9 ILP',
                statements: [
                  {
                    id: '387',
                    isLocked: false,
                    isRequired: false,
                    name: 'Take an interest survey or career assessment and record results in your Individual Learning Plan.',
                    step: 1,
                    evidences: [],
                    question: null,
                    results: [],
                    comments: [],
                    __typename: 'PlanGroupStatement',
                  },
                ],
                step: 2,
                __typename: 'PlanGroup',
              },
            ],
            id: '48',
            name: 'KY Grade 9 ILP',
            __typename: 'Plan',
            progress: {
              completed: 0,
              total: 1,
              __typename: 'PlanProgress',
            },
          },
          studentId: '10458930',
          __typename: 'Portfolio',
        },
        uuid: '123',
        __typename: 'Student',
      },
    },
  },
};

const plansMock: MockedResponse<TPortfolioPlansData> = {
  request: {
    query: PORTFOLIO_PLANS_QUERY,
  },
  result: {
    data: {
      portfolio: {
        plans: [
          {
            id: '48',
            name: 'Defined Portrait of a Graduate',
            evaluation: {
              id: '3',
            },
          },
          {
            id: '2',
            name: 'KY Grade 9 ILP',
            evaluation: {
              id: '1',
            },
          },
        ],
        studentId: '123',
      },
    },
  },
};

const studentPlansMock: MockedResponse = {
  request: {
    query: STUDENT_PORTFOLIO_PLANS,
    variables: { uuid: '123' },
  },
  result: {
    data: {
      student: {
        portfolio: {
          plans: [
            {
              id: '48',
              name: 'KY Grade 9 ILP',
              evaluation: {
                id: '2570',
                __typename: 'Evaluation',
              },
              __typename: 'Plan',
            },
          ],
          studentId: '10458930',
          __typename: 'Portfolio',
        },
        uuid: 'ffb8ccb6-ff90-48ce-8603-bec636ee5aa5',
        firstName: 'cleverteacher1',
        lastName: 'Demo',
        email: null,
        username: 'cleverteacher1_demo',
        __typename: 'Student',
      },
    },
  },
};

const planWithEvaluationResultMock = {
  data: {
    plan: {
      description:
        '<p><img style="display: block; margin-left: auto; margin-right: auto;" src="https://picsum.photos/600/300" width="795" height="57" /></p>\n<p>Defined has collaboratively constructed our Portrait of a Graduate which identifies competencies that can be practiced and supported through the utilization of Defined Learning and Defined Careers. This profile serves as our vision for teaching and learning and reinforces our beliefs through the performance tasks created and instructional processes supported.</p>',
      evaluation: {
        id: '3',
      },
      groups: [
        {
          description: '<p>Students are aware of and prepared for career opportunities.</p>',
          displayName: 'Career Readiness',
          id: '98',
          name: 'Defined - Career Readiness',
          statements: [
            {
              id: '267',
              name: 'Connects academic knowledge and skills to careers',
              step: 1,
              isLocked: false,
              comments: [],
              results: [],
              evidences: [],
              isRequired: false,
              question: null,
            },
            {
              id: '268',
              name: 'Shows awareness of opportunities and needs',
              step: 2,
              isLocked: false,
              comments: [],
              evidences: [],
              isRequired: false,
              results: [
                {
                  createdAt: '2023-09-09T07:28:43Z',
                  evaluator: {
                    firstName: 'Bruce',
                    lastName: 'Wayne',
                    uuid: 'someuuid',
                  },
                  result: EVALUATION_RESULTS_VALUES.IN_PROGRESS,
                },
              ],
              question: null,
            },
          ],
          step: 1,
        },
        {
          description: '<p>Students have met content standards for key subject areas.&nbsp;</p>',
          displayName: 'Academic Knowledge & Skills',
          id: '99',
          name: 'Defined - Academic Knowledge & Skills',
          evidences: [],
          isRequired: false,
          statements: [
            {
              id: '271',
              name: 'Demonstrates disciplinary understanding and use of competencies, practices, and processes',
              step: 1,
              isLocked: false,
              comments: [
                {
                  author: {
                    firstName: 'Bruce',
                    lastName: 'Wayne',
                    uuid: 'someuuid',
                  },
                  body: 'Could we add a status to my plan statement? It would help me track my progress more effectively. Thank you!',
                  createdAt: '2023-10-16T08:43:55Z',
                },
              ],
              results: [],
              evidences: [],
              isRequired: false,
              question: null,
            },
          ],
          step: 2,
        },
        {
          description:
            '<p>Students understand themselves, make good choices, and maintain relationships with others. &nbsp;</p>',
          displayName: 'Social & Emotional Learning',
          id: '100',
          name: 'Defined - Social & Emotional Learning',
          statements: [
            {
              id: '275',
              name: 'Accurately recognizes their own emotions, thoughts, and values',
              step: 1,
              isLocked: false,
              comments: [],
              results: [],
              evidences: [],
              isRequired: false,
              question: null,
            },
            {
              id: '276',
              name: 'Regulates emotions, thoughts, and behaviors',
              step: 2,
              isLocked: false,
              comments: [],
              results: [],
              evidences: [],
              isRequired: false,
              question: null,
            },
          ],
          step: 3,
        },
        {
          description:
            '<p>Students generate ideas, try new approaches, and are willing to take risks. &nbsp;</p>',
          displayName: 'Innovation & Creativity',
          id: '101',
          name: 'Defined - Innovation & Creativity',
          statements: [],
          step: 4,
        },
      ],
      id: '1',
      name: 'Defined Portrait of a Graduate',
      progress: {
        completed: 0,
        total: 5,
      },
    },
  },
};

const planWithEvaluationMock: MockedResponse = {
  request: {
    query: PLAN_WITH_EVALUATION_QUERY,
    variables: { id: '48' },
  },
  result: planWithEvaluationResultMock,
};
const renderComponent = (
  userInfo: UserInfo | StudentInfo = studentInfoMock.result.data.userInfo,
  type = 'student',
  mocks = [plansMock, planWithEvaluationMock]
) => {
  const isStudent = type === 'student';
  const routePath = isStudent ? '/plans' : '/students/:id/plans';
  const route = isStudent ? '/plans' : '/students/123/plans';

  return renderWithRouterAndReduxProvider(
    <MockedProvider mocks={mocks}>
      <NavigationContextProvider>
        <Route path={routePath}>
          <PortfolioPlans userInfo={userInfo} />
        </Route>
      </NavigationContextProvider>
    </MockedProvider>,
    { route: route, initialState: { session: { user: { type }, loginError: {} } } }
  );
};

describe('PortfolioPlans', () => {
  it('renders correctly', async () => {
    const { container } = renderComponent();

    const plansHeader = await screen.findByTestId('plans-header');
    expect(plansHeader).toHaveTextContent('Bruce Wayne');
    expect(plansHeader).toHaveTextContent('2 student plans');

    const planGroups = await screen.findAllByRole('listitem');
    expect(planGroups).toHaveLength(3);
    expect(planGroups[0]).toHaveTextContent('Career Readiness');
    expect(planGroups[1]).toHaveTextContent('Academic Knowledge & Skills');
    expect(planGroups[2]).toHaveTextContent('Social & Emotional Learning');

    expect(await screen.queryByText('Innovation & Creativity')).not.toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('displays plan statements after expanding plan group', async () => {
    renderComponent();

    const planGroupButton = await screen.findByRole('button', { name: /Career Readiness/ });
    userEvent.click(planGroupButton);

    const planGroup = screen.getByRole('listitem', { name: /Career Readiness/ });
    const planStatements = within(planGroup).getAllByRole('listitem');
    expect(planStatements).toHaveLength(2);

    expect(planStatements[0]).toHaveTextContent(
      'Connects academic knowledge and skills to careers'
    );
    expect(planStatements[0]).toHaveTextContent('Not started');
    expect(planStatements[1]).toHaveTextContent('Shows awareness of opportunities and needs');
    expect(planStatements[1]).toHaveTextContent('In progress');
  });

  it('displays plan statement with plan id and student id in url', async () => {
    renderComponent();

    const reportLink = await screen.findByRole('link', { name: 'Goals: Progress report' });

    expect(reportLink).toHaveAttribute('href', '/reports/student-progress/48');
  });
});

describe('PortfolioPlans as teacher', () => {
  it('renders correctly', async () => {
    const { container } = renderComponent(userInfoMock.result.data.userInfo, 'teacher', [
      studentPlansMock,
      studentPortfolioPlansWithEvaluationMock,
    ]);
    expect(container).toMatchSnapshot();
  });

  it('has student report link', async () => {
    renderComponent(userInfoMock.result.data.userInfo, 'teacher', [
      studentPlansMock,
      studentPortfolioPlansWithEvaluationMock,
    ]);

    const reportLink = await screen.findByRole('link', { name: 'Goals: Progress report' });

    expect(reportLink).toHaveAttribute('href', '/reports/student-progress/48/123');
  });
});
