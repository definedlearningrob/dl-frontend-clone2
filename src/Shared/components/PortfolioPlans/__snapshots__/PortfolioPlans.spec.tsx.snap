// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PortfolioPlans as teacher renders correctly 1`] = `
<div>
  <div
    class="flex flex-col gap-sm xxxl:gap-base"
  >
    <div
      class="flex gap-sm items-center bg-white rounded-sm py-sm px-base xxxl:py-base xxxl:px-md"
    >
      <div
        class="skeleton-rectangle skeletonRectangle !w-[44px] !h-[44px] rounded-full radius-md height-auto width-full-width darker animated"
      />
      <div
        class="flex flex-col gap-xxs flex-1"
      >
        <div
          class="skeleton-rectangle skeletonRectangle radius-md height-extra-small width-sm darker animated"
        />
        <div
          class="skeleton-rectangle skeletonRectangle radius-md height-auto width-sm darker animated"
        />
      </div>
    </div>
    <div
      class="card"
    >
      <div
        class="skeleton-rectangle skeletonRectangle mb-base xxxl:mb-md radius-sm height-large width-full-width darker animated"
      />
      <div
        class="py-xs px-sm xxxl:py-sm xxxl:px-md bg-neutral-200 rounded-sm xxxl:max-w-[1200px] mb-base"
      >
        <div
          class="flex gap-base"
        >
          <div
            class="flex items-center gap-xs"
          >
            <div
              class="flex"
            >
              <div
                class="flex align-center justify-center grow-0 shrink-0 bg-white border border-neutral-300 rounded-xs text-neutral-700 p-xxs"
                data-testid="icon"
              >
                <div
                  class="h-sm w-sm"
                />
              </div>
            </div>
            <span
              class="text-xs xxxl:text-sm font-medium leading-sm"
            >
              Not started
            </span>
          </div>
          <div
            class="flex items-center gap-xs"
          >
            <div
              class="flex"
            >
              <div
                class="flex align-center justify-center grow-0 shrink-0 bg-white border border-neutral-300 rounded-xs text-secondary-500 p-xxs"
                data-testid="icon"
              >
                <div
                  class="h-sm w-sm"
                />
              </div>
            </div>
            <span
              class="text-xs xxxl:text-sm font-medium leading-sm"
            >
              In progress
            </span>
          </div>
          <div
            class="flex items-center gap-xs"
          >
            <div
              class="flex"
            >
              <div
                class="flex align-center justify-center grow-0 shrink-0 bg-white border border-neutral-300 rounded-xs text-success-500 p-xxs"
                data-testid="icon"
              >
                <div
                  class="h-sm w-sm"
                />
              </div>
            </div>
            <span
              class="text-xs xxxl:text-sm font-medium leading-sm"
            >
              Completed
            </span>
          </div>
          <div
            class="flex items-center gap-xs"
          >
            <div
              class="flex"
            >
              <div
                class="flex align-center justify-center grow-0 shrink-0 bg-white border border-neutral-300 rounded-xs text-danger-500 p-xxs"
                data-testid="icon"
              >
                <div
                  class="h-sm w-sm"
                />
              </div>
            </div>
            <span
              class="text-xs xxxl:text-sm font-medium leading-sm"
            >
              Not met
            </span>
          </div>
          <div
            class="flex items-center gap-xs"
          >
            <div
              class="flex"
            >
              <div
                class="flex align-center justify-center grow-0 shrink-0 bg-white border border-neutral-300 rounded-xs text-neutral-700 p-xxs"
                data-testid="icon"
              >
                <div
                  class="h-sm w-sm"
                />
              </div>
            </div>
            <span
              class="text-xs xxxl:text-sm font-medium leading-sm"
            >
              Locked
            </span>
          </div>
        </div>
      </div>
      <ul
        class="flex flex-col gap-sm xxxl:gap-base"
      >
        <div
          class="skeleton-rectangle skeletonRectangle !h-[100px] xxxl:max-w-[1200px] radius-sm height-auto width-full-width darker animated"
        />
        <div
          class="skeleton-rectangle skeletonRectangle !h-[100px] xxxl:max-w-[1200px] radius-sm height-auto width-full-width darker animated"
        />
        <div
          class="skeleton-rectangle skeletonRectangle !h-[100px] xxxl:max-w-[1200px] radius-sm height-auto width-full-width darker animated"
        />
      </ul>
    </div>
  </div>
</div>
`;

exports[`PortfolioPlans renders correctly 1`] = `
<div>
  <div
    class="flex flex-col gap-sm xxxl:gap-base"
  >
    <div
      class="flex gap-sm justify-between bg-white rounded-sm py-sm px-base xxxl:py-base xxxl:px-md"
      data-testid="plans-header"
    >
      <div
        class="flex gap-sm"
      >
        <div
          class="recharts-wrapper"
          role="region"
          style="position: relative; cursor: default; width: 44px; height: 44px;"
        >
          <svg
            class="recharts-surface"
            cx="50%"
            cy="50%"
            height="44"
            style="width: 100%; height: 100%;"
            viewBox="0 0 44 44"
            width="44"
          >
            <title />
            <desc />
            <defs>
              <clippath
                id="recharts1-clip"
              >
                <rect
                  height="44"
                  width="44"
                  x="0"
                  y="0"
                />
              </clippath>
            </defs>
            <g
              class="recharts-layer recharts-polar-angle-axis"
            >
              <path
                class="recharts-polygon angleAxis"
                cx="22"
                cy="22"
                d="M22,-4.399999999999999L37.51753066052129,0.6419513485013901L47.10789203019205,13.84195134850139L47.10789203019205,30.158048651498614L37.51753066052129,43.358048651498606L22,48.4L6.482469339478714,43.358048651498606L-3.107892030192051,30.158048651498614L-3.1078920301920547,13.841951348501393L6.4824693394787065,0.6419513485013901L21.999999999999996,-4.399999999999999L22,-4.399999999999999Z"
                fill="none"
                orientation="outer"
                radius="26.4"
              />
              <g
                class="recharts-layer recharts-polar-angle-axis-ticks"
              >
                <g
                  class="recharts-layer recharts-polar-angle-axis-tick"
                >
                  <line
                    class="angleAxis"
                    cx="22"
                    cy="22"
                    fill="none"
                    orientation="outer"
                    radius="26.4"
                    x1="22"
                    x2="22.000000000000004"
                    y1="-4.399999999999999"
                    y2="-12.399999999999999"
                  />
                </g>
                <g
                  class="recharts-layer recharts-polar-angle-axis-tick"
                >
                  <line
                    class="angleAxis"
                    cx="22"
                    cy="22"
                    fill="none"
                    orientation="outer"
                    radius="26.4"
                    x1="37.51753066052129"
                    x2="42.219812678861075"
                    y1="0.6419513485013901"
                    y2="-5.8301846064981895"
                  />
                </g>
                <g
                  class="recharts-layer recharts-polar-angle-axis-tick"
                >
                  <line
                    class="angleAxis"
                    cx="22"
                    cy="22"
                    fill="none"
                    orientation="outer"
                    radius="26.4"
                    x1="47.10789203019205"
                    x2="54.71634416055328"
                    y1="13.84195134850139"
                    y2="11.36981539350181"
                  />
                </g>
                <g
                  class="recharts-layer recharts-polar-angle-axis-tick"
                >
                  <line
                    class="angleAxis"
                    cx="22"
                    cy="22"
                    fill="none"
                    orientation="outer"
                    radius="26.4"
                    x1="47.10789203019205"
                    x2="54.71634416055328"
                    y1="30.158048651498614"
                    y2="32.63018460649819"
                  />
                </g>
                <g
                  class="recharts-layer recharts-polar-angle-axis-tick"
                >
                  <line
                    class="angleAxis"
                    cx="22"
                    cy="22"
                    fill="none"
                    orientation="outer"
                    radius="26.4"
                    x1="37.51753066052129"
                    x2="42.219812678861075"
                    y1="43.358048651498606"
                    y2="49.83018460649819"
                  />
                </g>
                <g
                  class="recharts-layer recharts-polar-angle-axis-tick"
                >
                  <line
                    class="angleAxis"
                    cx="22"
                    cy="22"
                    fill="none"
                    orientation="outer"
                    radius="26.4"
                    x1="22"
                    x2="22.000000000000004"
                    y1="48.4"
                    y2="56.4"
                  />
                </g>
                <g
                  class="recharts-layer recharts-polar-angle-axis-tick"
                >
                  <line
                    class="angleAxis"
                    cx="22"
                    cy="22"
                    fill="none"
                    orientation="outer"
                    radius="26.4"
                    x1="6.482469339478714"
                    x2="1.7801873211389285"
                    y1="43.358048651498606"
                    y2="49.83018460649819"
                  />
                </g>
                <g
                  class="recharts-layer recharts-polar-angle-axis-tick"
                >
                  <line
                    class="angleAxis"
                    cx="22"
                    cy="22"
                    fill="none"
                    orientation="outer"
                    radius="26.4"
                    x1="-3.107892030192051"
                    x2="-10.716344160553277"
                    y1="30.158048651498614"
                    y2="32.63018460649819"
                  />
                </g>
                <g
                  class="recharts-layer recharts-polar-angle-axis-tick"
                >
                  <line
                    class="angleAxis"
                    cx="22"
                    cy="22"
                    fill="none"
                    orientation="outer"
                    radius="26.4"
                    x1="-3.1078920301920547"
                    x2="-10.716344160553284"
                    y1="13.841951348501393"
                    y2="11.369815393501813"
                  />
                </g>
                <g
                  class="recharts-layer recharts-polar-angle-axis-tick"
                >
                  <line
                    class="angleAxis"
                    cx="22"
                    cy="22"
                    fill="none"
                    orientation="outer"
                    radius="26.4"
                    x1="6.4824693394787065"
                    x2="1.7801873211389214"
                    y1="0.6419513485013901"
                    y2="-5.830184606498186"
                  />
                </g>
                <g
                  class="recharts-layer recharts-polar-angle-axis-tick"
                >
                  <line
                    class="angleAxis"
                    cx="22"
                    cy="22"
                    fill="none"
                    orientation="outer"
                    radius="26.4"
                    x1="21.999999999999996"
                    x2="21.999999999999993"
                    y1="-4.399999999999999"
                    y2="-12.399999999999999"
                  />
                </g>
              </g>
            </g>
            <circle
              class="fill-neutral-200"
              cx="22"
              cy="22"
              r="18"
            />
            <g
              class="recharts-layer recharts-area fill-neutral-400"
            >
              <g
                class="recharts-layer recharts-radial-bar-background"
              >
                <path
                  class="recharts-sector recharts-radial-bar-background-sector"
                  cx="22"
                  cy="22"
                  d="M 22,-0.14999999999999858
    A 22.15,22.15,0,
    1,1,
    21.999613409570706,-0.14999999662635943
  L 21.99968322274078,3.8500000027644035
            A 18.15,18.15,0,
            1,0,
            22,3.8500000000000014 Z"
                  fill="#B0B5C9"
                  fill-opacity="1"
                  name="progress"
                  role="img"
                />
              </g>
              <g
                class="recharts-layer recharts-radial-bar-sectors"
              >
                <g
                  class="recharts-layer"
                />
              </g>
            </g>
            <text
              class="text-xxs font-medium fill-font-secondary"
              dominant-baseline="middle"
              text-anchor="middle"
              x="22"
              y="24"
            >
              0/5
            </text>
          </svg>
        </div>
        <div>
          <h4
            class="text-base xxxl:text-lg mb-xxs"
          >
            Goals: Plan
          </h4>
          <div
            class="flex items-center gap-xxs xxxl:gap-xs text-xxs xxxl:text-xs leading-lg"
          >
            <span
              class="text-primary-500 font-medium"
            >
              Bruce Wayne
            </span>
            <div
              class="w-xxs h-xxs rounded-full bg-neutral-300"
            />
            <span>
              2 student plans
            </span>
          </div>
        </div>
      </div>
      <div
        class="flex items-center gap-xs xxxl:gap-sm"
      >
        <label
          class="flex flex-col leading-lg min-w-[120px] gap-xxxs"
        >
          <div
            class="w-[240px] xxxl:w-[320px] css-b62m3t-container"
          >
            <span
              class="css-1f43avz-a11yText-A11yText"
              id="react-select-2-live-region"
            />
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              class="css-1f43avz-a11yText-A11yText"
              role="log"
            />
            <div
              class="!rounded-sm text-font-secondary !cursor-pointer outline-none !shadow-none !min-h-0 !border-neutral-300 hover:!border-neutral-400 !bg-white select__control css-13cymwt-control"
            >
              <div
                class="flex gap-xxs items-center !flex-nowrap !text-xxs !p-xs !pr-0 !py-xxs h-[30px] select__value-container select__value-container--has-value css-1fdsijx-ValueContainer"
              >
                <div
                  class="!m-0 !leading-[22px] h-[22px] select__single-value css-1dimb5e-singleValue"
                >
                  Defined Portrait of a Graduate
                </div>
                <div
                  class="!p-0 !m-0 !leading-[22px] h-[22px] select__input-container css-qbdosj-Input"
                  data-value=""
                >
                  <input
                    aria-activedescendant=""
                    aria-autocomplete="list"
                    aria-expanded="false"
                    aria-haspopup="true"
                    autocapitalize="none"
                    autocomplete="off"
                    autocorrect="off"
                    class="select__input"
                    id="react-select-2-input"
                    role="combobox"
                    spellcheck="false"
                    style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                    tabindex="0"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class="!p-0 !pr-xs select__indicators css-1hb7zxy-IndicatorsContainer"
              >
                <div
                  aria-hidden="true"
                  class="!p-0 select__indicator select__dropdown-indicator css-1xc3v61-indicatorContainer"
                >
                  <div
                    class="flex"
                  >
                    <div
                      class="flex align-center justify-center grow-0 shrink-0 text-neutral-400 text-neutral-800"
                      data-testid="icon"
                    >
                      <div
                        class="h-sm w-sm"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </label>
        <a
          href="/reports/student-progress/48"
        >
          <button
            class="button flex items-center justify-center gap-xs rounded-sm cursor-pointer text-sm font-medium no-underline focus-visible:outline focus-visible:outline-1 focus-visible:outline-offset-[3px] -primary bg-primary-500 hover:bg-primary-600 outline-primary-500 border-primary-500 py-xs px-sm !text-xs leading-[14px] !gap-xxs border text-white"
            data-testid="button"
            type="button"
          >
            <span
              class="items-center flex justify-center gap-xs"
            >
              Goals: Progress report
            </span>
          </button>
        </a>
      </div>
    </div>
    <div
      class="card"
    >
      <div
        class="injected-content mb-base xxxl:mb-md text-xs xxxl:text-sm"
      >
        <p>
          <img
            height="57"
            src="https://picsum.photos/600/300"
            style="display: block; margin-left: auto; margin-right: auto;"
            width="795"
          />
        </p>
        

        <p>
          Defined has collaboratively constructed our Portrait of a Graduate which identifies competencies that can be practiced and supported through the utilization of Defined Learning and Defined Careers. This profile serves as our vision for teaching and learning and reinforces our beliefs through the performance tasks created and instructional processes supported.
        </p>
      </div>
      <div
        class="py-xs px-sm xxxl:py-sm xxxl:px-md bg-neutral-200 rounded-sm xxxl:max-w-[1200px] mb-base"
      >
        <div
          class="flex gap-base"
        >
          <div
            class="flex items-center gap-xs"
          >
            <div
              class="flex"
            >
              <div
                class="flex align-center justify-center grow-0 shrink-0 bg-white border border-neutral-300 rounded-xs text-neutral-700 p-xxs"
                data-testid="icon"
              >
                <div
                  class="h-sm w-sm"
                />
              </div>
            </div>
            <span
              class="text-xs xxxl:text-sm font-medium leading-sm"
            >
              Not started
            </span>
          </div>
          <div
            class="flex items-center gap-xs"
          >
            <div
              class="flex"
            >
              <div
                class="flex align-center justify-center grow-0 shrink-0 bg-white border border-neutral-300 rounded-xs text-secondary-500 p-xxs"
                data-testid="icon"
              >
                <div
                  class="h-sm w-sm"
                />
              </div>
            </div>
            <span
              class="text-xs xxxl:text-sm font-medium leading-sm"
            >
              In progress
            </span>
          </div>
          <div
            class="flex items-center gap-xs"
          >
            <div
              class="flex"
            >
              <div
                class="flex align-center justify-center grow-0 shrink-0 bg-white border border-neutral-300 rounded-xs text-success-500 p-xxs"
                data-testid="icon"
              >
                <div
                  class="h-sm w-sm"
                />
              </div>
            </div>
            <span
              class="text-xs xxxl:text-sm font-medium leading-sm"
            >
              Completed
            </span>
          </div>
          <div
            class="flex items-center gap-xs"
          >
            <div
              class="flex"
            >
              <div
                class="flex align-center justify-center grow-0 shrink-0 bg-white border border-neutral-300 rounded-xs text-danger-500 p-xxs"
                data-testid="icon"
              >
                <div
                  class="h-sm w-sm"
                />
              </div>
            </div>
            <span
              class="text-xs xxxl:text-sm font-medium leading-sm"
            >
              Not met
            </span>
          </div>
          <div
            class="flex items-center gap-xs"
          >
            <div
              class="flex"
            >
              <div
                class="flex align-center justify-center grow-0 shrink-0 bg-white border border-neutral-300 rounded-xs text-neutral-700 p-xxs"
                data-testid="icon"
              >
                <div
                  class="h-sm w-sm"
                />
              </div>
            </div>
            <span
              class="text-xs xxxl:text-sm font-medium leading-sm"
            >
              Locked
            </span>
          </div>
        </div>
      </div>
      <ul
        class="flex flex-col gap-sm xxxl:gap-base"
      >
        <li
          aria-label="Career Readiness"
          class="overflow-hidden group p-base xxxl:p-md rounded-sm bg-white xxxl:max-w-[1200px] transition-all border border-neutral-300 focus-within:border-primary-500 plan-group hover:bg-neutral-200 hover:border-neutral-400"
          data-id="98"
        >
          <div
            class="relative"
            role="button"
          >
            <div
              class="flex gap-xs xxxl:gap-sm mb-xs xxxl:mb-sm mr-lg"
            >
              <div
                class="flex"
              >
                <div
                  class="flex align-center justify-center grow-0 shrink-0 outline outline-1 outline-neutral-300 bg-white rounded-xs !h-fit text-secondary-500 p-xxs"
                  data-testid="icon"
                >
                  <div
                    class="h-sm w-sm"
                  />
                </div>
              </div>
              <h5
                class="mb-0 text-sm xxxl:text-base leading-lg max-w-[680px] xxxl:max-w-[840px]"
              >
                Career Readiness
              </h5>
            </div>
            <button
              class="icon-button -size-sm transition-colors !bg-white xxxl:!p-xxs !border !border-solid !rounded-xs !border-white focus-visible:!border-primary-500 absolute right-0 top-0 group-hover:text-primary-500"
              data-testid="icon-button"
              type="button"
            >
              <div
                class="icon icon-button__icon -sm-size"
                data-testid="icon"
              >
                <div />
              </div>
            </button>
            <div
              class="text-neutral-700 ml-md xxxl:ml-lg max-w-[680px] xxxl:max-w-[840px] text-xs xxxl:text-sm [&_p:last-child]:mb-0"
            >
              <p>
                Students are aware of and prepared for career opportunities.
              </p>
            </div>
          </div>
        </li>
        <li
          aria-label="Academic Knowledge & Skills"
          class="overflow-hidden group p-base xxxl:p-md rounded-sm bg-white xxxl:max-w-[1200px] transition-all border border-neutral-300 focus-within:border-primary-500 plan-group hover:bg-neutral-200 hover:border-neutral-400"
          data-id="99"
        >
          <div
            class="relative"
            role="button"
          >
            <div
              class="flex gap-xs xxxl:gap-sm mb-xs xxxl:mb-sm mr-lg"
            >
              <div
                class="flex"
              >
                <div
                  class="flex align-center justify-center grow-0 shrink-0 outline outline-1 outline-neutral-300 bg-white rounded-xs !h-fit text-font-secondary p-xxs"
                  data-testid="icon"
                >
                  <div
                    class="h-sm w-sm"
                  />
                </div>
              </div>
              <h5
                class="mb-0 text-sm xxxl:text-base leading-lg max-w-[680px] xxxl:max-w-[840px]"
              >
                Academic Knowledge & Skills
              </h5>
            </div>
            <button
              class="icon-button -size-sm transition-colors !bg-white xxxl:!p-xxs !border !border-solid !rounded-xs !border-white focus-visible:!border-primary-500 absolute right-0 top-0 group-hover:text-primary-500"
              data-testid="icon-button"
              type="button"
            >
              <div
                class="icon icon-button__icon -sm-size"
                data-testid="icon"
              >
                <div />
              </div>
            </button>
            <div
              class="text-neutral-700 ml-md xxxl:ml-lg max-w-[680px] xxxl:max-w-[840px] text-xs xxxl:text-sm [&_p:last-child]:mb-0"
            >
              <p>
                Students have met content standards for key subject areas. 
              </p>
            </div>
          </div>
        </li>
        <li
          aria-label="Social & Emotional Learning"
          class="overflow-hidden group p-base xxxl:p-md rounded-sm bg-white xxxl:max-w-[1200px] transition-all border border-neutral-300 focus-within:border-primary-500 plan-group hover:bg-neutral-200 hover:border-neutral-400"
          data-id="100"
        >
          <div
            class="relative"
            role="button"
          >
            <div
              class="flex gap-xs xxxl:gap-sm mb-xs xxxl:mb-sm mr-lg"
            >
              <div
                class="flex"
              >
                <div
                  class="flex align-center justify-center grow-0 shrink-0 outline outline-1 outline-neutral-300 bg-white rounded-xs !h-fit text-font-secondary p-xxs"
                  data-testid="icon"
                >
                  <div
                    class="h-sm w-sm"
                  />
                </div>
              </div>
              <h5
                class="mb-0 text-sm xxxl:text-base leading-lg max-w-[680px] xxxl:max-w-[840px]"
              >
                Social & Emotional Learning
              </h5>
            </div>
            <button
              class="icon-button -size-sm transition-colors !bg-white xxxl:!p-xxs !border !border-solid !rounded-xs !border-white focus-visible:!border-primary-500 absolute right-0 top-0 group-hover:text-primary-500"
              data-testid="icon-button"
              type="button"
            >
              <div
                class="icon icon-button__icon -sm-size"
                data-testid="icon"
              >
                <div />
              </div>
            </button>
            <div
              class="text-neutral-700 ml-md xxxl:ml-lg max-w-[680px] xxxl:max-w-[840px] text-xs xxxl:text-sm [&_p:last-child]:mb-0"
            >
              <p>
                Students understand themselves, make good choices, and maintain relationships with others.  
              </p>
            </div>
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
